### PR TITLE
Timeout

### PR DESCRIPTION
**Pull request by kornelski**
(requires #158)

Short timeouts didn't work well with large images, because the timeout was checked rarely.

This adds more checks of `deadline.passed()`

### CODE REVIEW

**Comment by kornelski** [comment]
Rebased

**Comment by shssoichiro** [comment]
Sorry, one more from merging #162. :)

File: src/lib.rs [hunk 1]
```diff
@@ -341,6 +341,8 @@ pub fn optimize(input: &InFile, output: &OutFile, opts: &Options) -> PngResult<(
         eprintln!("Processing: {}", input);
     }
 
+    let deadline = Arc::new(Deadline::new(opts.timeout, opts.verbosity.is_some()));
```
Comment (shssoichiro) [inline src/lib.rs]: Does it make more sense to instantiate the Deadline (here and on 443) inside `optimize_png` to avoid the duplication?

Comment (kornelski) [reply src/lib.rs]: It starts counting time from when it's instantiated, and I wanted it to also include time it takes to load the image.

**Pull request merged**